### ISSUE TITLE
Calendar widget one to two days behind

### ISSUE THREAD

**Issue by ultimase** [Etar-Group/Etar-Calendar#1156]
Running calyxos, and noticed recently that the calendar widget appears to lag a day or two behind the actual day. For example today while being Tuesday the widget is still showing Sunday events. This is occurring on two of my phones. Anyone else stumble onto this?

**Comment by kuba1909** [comment]
I noticed the same issue on the etar calendar shipped by my official lineage os rom. I always have to remove and add the widget back again, which is very inconvenient.

**Comment by jewelooper** [comment]
Same problem here with calyxos 3.5.1 and etar 1.0.29.
The widget does not update to current date. Only when readding the widget to home screen the widget shows the correct date again.

**Comment by kuba1909** [comment]
I manged to get rid of the problem by visiting etars app-info and

- under permissions I unticked the checkbox for unused apps (which possibly removes the granted permissions after a period of time)
- under battery I removed the battery usage limitation (I guess this one did actually the trick)

Also I noted, that etars widget isn't the only one with that issue but other widgets are affected aswell. Although there are some widgets, which are not affected…

**Comment by Forage** [comment]
I can confirm that changing the settings mentioned by kuba1909 appears to do the trick. The date of the widget moved one day forward this night.

**Comment by jondo** [comment]
Allowing unlimited battery use also solved this for me. Thanks, @kuba1909 !

**Comment by friedger** [comment]
Instead of unlimited battery use permissions, it might be better to use a scheduler or so that updates the widget once a day
Or add a guesture/UI to update it manually

**Comment by hcgrove** [comment]
@kuba1909 I have1
My comments to those hints:
- You can't really lock a widget, but the main etar app is locked. I haven't experienced any battery settings (the ones I've found at least) to be reset, so I'm not sure that is particularly relevant either.
- The "Battery optimization" section seems to be based on an older version of OxygenOS, on my phone (I guess it's because of the OxygenOS version) there is no "Gear Icon" when I go to "Apps" in "Settings".
- There is no "Battery optimization" in "System settings > Battery" on my OnePlus11. And I haven't found anything called "Advanced optimization" when looking around in the battery settings. I did just turn off "Sleep standby optimization", I don't know why I hadn't done that before.
- "Recent apps clearing behaviour" I can't find those settings, but the main etar app (again it's the widget that gives problems and I never swipe that away - don't even know how to) seems to remain in my recent app list, so possibly not so relevant.

**Comment by hcgrove** [comment]
This morning (and having turned that setting off, the phone should no longer try to figure out when I'm sleeping and optiimise battery usage then) the widget said it was wednesday, when it is thursday. I.e. te widget is still not getting updated, I've tried every relevant setting I can find on my OnePlus11 running OxygenOS 13, and the hints on [](https://dontkillmyapp.com/oneplus) seem to be for an older version of OxygenOS.

**Comment by hcgrove** [comment]
What I wrote above was not completed. In addition to "Allow background activity" as mentioned above (if anybody needs to find it, it's in "Settings" (app)/"Apps"/"App management"/"Etar"/"Battery usage" - but see below) I had also chosen never to optimise (can't find the setting in the new version of OxygenOS can came a few days after the change I mention below) Etar's battery usage.

Some weeks ago I tried turning off "Allow background activity", that enabled the setting called "Allow foreground activity"  just above, that was dimmed out before but seemed to be so in the "on" position. I didn't change that setting just left it "on" - and since then etar's widget has worked as it should. 
As I would characterise the widget's operation as foreground activity, I doubt it was actually allowed (even though it seemed so), but in any case, but in any case it's strange that disallowing anything made stuff work.
A couple of days later (when I first started to think it might work), I made a test event with a notification set, and wasn't warned, but earlier today I got a notification for an event (that was created on 26 June), so I'm cautiously optimistic, I've just added an notification for an event on monday (but the notification is set to happen 1 day before, i.e. on sunday) to see if it works.

TL;DR: It might help to turn off certain settings you think should help by allowing certain kinds of activity.
